File: python/Sublist.py
"""
This exercise stub and the test suite contain several enumerated constants.

Enumerated constants can be done with a NAME assigned to an arbitrary,
but unique value. An integer is traditionally used because it’s memory
efficient.
It is a common practice to export both constants and functions that work with
those constants (ex. the constants in the os, subprocess and re modules).

You can learn more here: https://en.wikipedia.org/wiki/Enumerated_type
"""

# Possible sublist categories.
# Change the values as you see fit.
SUBLIST = -1
SUPERLIST = 1
EQUAL = 0
UNEQUAL = 2
def is_sublist(short, long):
    """Return True if 'short' is a contiguous sublist of 'long'."""
    n, m = len(short), len(long)
    if n == 0:
        return True
    for i in range(m - n + 1):
        if long[i:i+n] == short:
            return True
    return False
def sublist(list_one, list_two):
    if list_one == list_two:
        return EQUAL
    if is_sublist(list_one, list_two):
        return SUBLIST
    if is_sublist(list_two, list_one):
        return SUPERLIST
    return UNEQUAL

File: javascript/Sublist.js
export class List {
  constructor(newList) {
    this.array = newList || [];
  }
  
  compare(listTwoObj) {
    let result = undefined
    let sublistStart = 0
    const listOne = this.array
    const listTwo = listTwoObj.array
    const listOneJoined = listOne.join()
    const listTwoJoined = listTwo.join()
    if(listOne.length < listTwo.length){
      if(listTwoJoined.includes(listOneJoined)){
        result = "SUBLIST"
      }
      else result = "UNEQUAL"
    }
    else if(listTwo.length < listOne.length){
      if(listOneJoined.includes(listTwoJoined)){
        result = "SUPERLIST"
      }
      else result = "UNEQUAL"
    }
    else if(listOne.length === listTwo.length){
      if(listOne.length === 0 && listTwo.length === 0){
        result = "EQUAL"
      }
      listOne.forEach((listItem, idx) => {
        if(listItem === listTwo[idx]){
          result = "EQUAL"
        }
        else {
          result = "UNEQUAL"
        }
      })
    }
      return result
    }
  }